Move bounding box out of the node data

dalg24 opened the pull request:
DO NOT MERGE

dalg24 commented on packages/Search/src/DTK_LinearBVH_decl.hpp:
```diff
@@ -88,13 +88,13 @@ class BoundingVolumeHierarchy
     bounding_volume_type const &getBoundingVolume( Node const *node ) const
     {
-        return node->bounding_box;
+        return _bounding_volumes( node - getRoot() );
```
OK

dalg24 commented on packages/Search/src/details/DTK_DetailsTreeConstruction_def.hpp:
```diff
@@ -268,8 +273,7 @@ void TreeConstruction<DeviceType>::initializeLeafNodes(
         Kokkos::RangePolicy<ExecutionSpace>( 0, n ), KOKKOS_LAMBDA( int i ) {
-            // FIXME: Here also accessing node bounding box data member
-            leaf_nodes( i ).bounding_box = bounding_boxes( indices( i ) );
+            bounding_volumes( i + n - 1 ) = bounding_boxes( indices( i ) );
```
Good point :)

aprokop commented:
> In principle we could template `TreeConstruction` on the tree type.

I guess. However tree construction is too tightly coupled to a specific tree.

dalg24 commented:
> > In principle we could template `TreeConstruction` on the tree type.
> 
> I guess. However tree construction is too tightly coupled to a specific tree.

The intent is not to use with another type, it is to be able to deduce the device type and the bounding volume type.

dalg24 commented:
@aprokop I addressed your comments.  Should we close that PR and put it in the backlog?
For reference, these changes slightly improve performance on the GPU (< 2-3% for `BVH::query()`) but degrade it on the CPU (few percents)

aprokop commented:
> Should we close that PR and put it in the backlog?

Yes. Create an issue (if we don't have it) to keep track of the performance PRs.
